middleware/proxy/upstream.go

```diff
@@ -56,17 +57,38 @@ func NewStaticUpstreams(c parse.Dispenser) ([]Upstream, error) {
 		if !c.Args(&upstream.from) {
 			return upstreams, c.ArgErr()
 		}
-		to := c.RemainingArgs()
-		if len(to) == 0 {
-			return upstreams, c.ArgErr()
+
+		var to []string
+		for _, t := range c.RemainingArgs() {
+			parsed, err := parseUpstream(t)
+			if err != nil {
+				return upstreams, err
+			}
+			to = append(to, parsed...)
 		}
 
 		for c.NextBlock() {
-			if err := parseBlock(&c, upstream); err != nil {
-				return upstreams, err
+			switch c.Val() {
+			case "upstream":
+				if !c.NextArg() {
+					return upstreams, c.ArgErr()
+				}
+				parsed, err := parseUpstream(c.Val())
+				if err != nil {
+					return upstreams, err
+				}
+				to = append(to, parsed...)
+			default:
+				if err := parseBlock(&c, upstream); err != nil {
+					return upstreams, err
+				}
 			}
 		}
 
+		if len(to) == 0 {
+			return upstreams, c.ArgErr()
+		}
+
 		upstream.Hosts = make([]*UpstreamHost, len(to))
```

Comment:
My intuition tells me it is, but have you checked that it's to leave `upstream.Hosts` nil in the case `len(to) == 0` (just above)?